wallet/db.c

```diff
@@ -503,6 +565,60 @@ bool sqlite3_bind_pubkey(sqlite3_stmt *stmt, int col, const struct pubkey *pk)
 	return true;
 }
 
+bool sqlite3_bind_pubkey_array(sqlite3_stmt *stmt, int col,
+			       const struct pubkey *pks)
+{
+	const tal_t *tmpctx;
+	size_t n;
+	size_t i;
+	u8 *ders;
+
+	if (!pks) {
+		sqlite3_bind_null(stmt, col);
+		return true;
+	}
+
+	tmpctx = tal_tmpctx(pks);
+	n = tal_count(pks);
+	ders = tal_arr(tmpctx, u8, n * PUBKEY_DER_LEN);
+
+	for (i = 0; i < n; ++i)
+		pubkey_to_der(&ders[i * PUBKEY_DER_LEN], &pks[i]);
+	sqlite3_bind_blob(stmt, col, ders, tal_len(ders), SQLITE_TRANSIENT);
+
+	tal_free(tmpctx);
+	return true;
+}
+struct pubkey *sqlite3_column_pubkey_array(const tal_t *ctx,
+					   sqlite3_stmt *stmt, int col)
+{
+	const tal_t *tmpctx;
+	size_t i;
+	size_t n;
+	struct pubkey *ret;
+	const u8 *ders;
+
+	if (sqlite3_column_type(stmt, col) == SQLITE_NULL)
+		return NULL;
+
+	tmpctx = tal_tmpctx(ctx);
+	n = sqlite3_column_bytes(stmt, col) / PUBKEY_DER_LEN;
+	assert(n * PUBKEY_DER_LEN == sqlite3_column_bytes(stmt, col));
+	ret = tal_arr(tmpctx, struct pubkey, n);
+	ders = sqlite3_column_blob(stmt, col);
+
+	for (i = 0; i < n; ++i) {
+		if (!pubkey_from_der(&ders[i * PUBKEY_DER_LEN], PUBKEY_DER_LEN, &ret[i])) {
+			tal_free(tmpctx);
+			return NULL;
+		}
+	}
+
+	ret = tal_steal(ctx, ret);
+	tal_free(tmpctx);
+	return ret;
```

Comment:
Stylistic issue: tmpctx is actually unnecessary here.  You could just allocate ret off ctx, then where you return NULL, instead do `return tal_free(ret);` which is a cute and useful one-liner to clean up and return NULL.